qiskit/circuit/barrier.py

```diff
@@ -21,7 +21,10 @@
 
 
 class Barrier(Instruction):
-    """Barrier instruction."""
+    """A barrier is a visual indicator of the grouping of a circuit section. 
+    Actings as a directive for circuit compilation to separate pieces 
```

Comment:
```suggestion
    It also acts as a directive for circuit compilation to separate pieces 
```

Reply:
Thanks for the note, I have committed your suggestion!